File: tp-image-kit/tpimage.c
/*
 
 ######  #     #   ###     #       #     ######
 #       ##    #  #   #                  #
 #       # #   #   #      ##      ##     #
 #####   #  #  #    #      #       #     #####
 #       #   # #     #     #       #     #
 #       #    ##  #   #    #       #     #
 ######  #     #   ###   #####   #####   ######    
 
 # 2017-2018
 # ---------
 
 */


#include <stdio.h>
#include <stdlib.h>

#include <sys/time.h>

#include "pnm.h"
#include "effets_photom.h"
#include "effets_geom.h"
#include "filtrages.h"

/*!
 \file tpimage.c
 \author
 \date
 \brief programme principal : test des fonctionnalités de manipulation d'image
 */


int main(int argc, char **argv)
{
    int xsize, /*! largeur de l'image */
    ysize;     /*! hauteur de l'image */
    PnmFile fich; /*! "fichier" image : nom, type, taille, bitmap */
    PnmFile fich2; /*! "fichier" image : nom, type, taille, bitmap */
    unsigned char * res=NULL;
    char cmd[256];
    
    if (argc<3) {
        fprintf(stderr,"usage : %s fichier.p[b|g|p]m\n fichier.p[b|g|p]m\n",argv[0]);
        exit(1);
    }
    srand(time(NULL)); /* initialisation du générateur pseudo-aléatoire */
    
    fich.name=argv[1];
    lirefich(&fich);
    fprintf(stderr,"Lecture OK (H=%d L=%d)\n",fich.ysize,fich.xsize);
    xsize=fich.xsize;
    ysize=fich.ysize;
    if ((fich.type!=2)&&(fich.type!=5)) exit(1);

    sprintf(cmd,"open %s -a ", argv[1]);
    sprintf(cmd,"atril %s &", argv[1]);
    system(cmd);

    res = inversion_videoNB(fich.bitmap, xsize, ysize);
    
    fich2.bitmap=res;
    fich2.xsize=xsize;
    fich2.ysize=ysize;
    fich2.maxvalue=255;
    fich2.name=argv[2];
    fich2.type=5;
    sauverfich(&fich2);
    
    sprintf(cmd,"atril %s &", argv[2]);
    system(cmd);
    
    return 0;
}

File: tp-image-kit/tpimage.py
#-*- coding: utf-8 -*-

######  #     #   ###     #       #     ######
#       ##    #  #   #                  #
#       # #   #   #      ##      ##     #
#####   #  #  #    #      #       #     #####
#       #   # #     #     #       #     #
#       #    ##  #   #    #       #     #
######  #     #   ###   #####   #####   ######

# 2017-2018
# ---------

import sys
import random

from PIL.Image import *


from effets_photom import *
from effets_geom import *
from filtrages import *


# /*!
#  \file tpimage.py
#  \author Pierre Tellier
#  \date  16/10/2017
#  \brief programme principal : test des fonctionnalités de manipulation d'image
# */



def main():
    argc=len(sys.argv)
    if argc<3 :
        print "usage : ", sys.argv[0], " fichier.png fichier.png"
        return
    
    random.seed()  #initialisation du générateur pseudo-aléatoire

    img1=open(sys.argv[1])
    (xsize,ysize) = img1.size
    print "Lecture OK ( H =", ysize, " L =", xsize,")"
    Image.show(img1)

    img2 = inversion_videoNB(img1, xsize, ysize)

    Image.show(img2)
    img2.save(sys.argv[2])

    img1.close()
    img2.close()


if __name__ == "__main__":
    main()





